backend/core/src/listings/listings.service.ts

```diff
@@ -137,6 +162,66 @@ export class ListingsService {
     if (!result) {
       throw new NotFoundException()
     }
+    if (lang !== Language.en) {
+      // Get key-value pairs from listing to be translated
+      const translations = this.getTranslations(result)
+
+      const translatedValues = await this.translateService().translate(translations.values, {
+        from: Language.en,
+        to: lang,
+      })
+      // Attach translated values to the listing
+      translations.keys.forEach((key, index) => {
+        this.setValue(result, key, translatedValues[0][index])
+      })
+    }
+
+    return result
+  }
+
+  private translateService = () => {
```

Comment:
I like the approach you've laid out here, but I'm wondering if everything from this point to the end of the file should be abstracted out of the listings service into its own module or whatever. That way we could generically use Google Translate from anywhere we might want in the future (not just listings).

Reply:
@software-project ,
I agree with @jaredcwhite and you set this up in a way that should allow you to easily do that. Can you create a `/src/translation` dir that has a `translation.service.ts` file with the appropriate code transferred to it. getTranslations can continue to receive listing: Listing, but in the future, if we ever need it for anything else, we can update it to use a generic.